api4/team.go

```diff
@@ -1405,6 +1428,9 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 	auditRec.AddMeta("emails", emailList)
 
 	if len(memberInvite.ChannelIds) > 0 {
+		// Check if the user sending the invitation has access to the channels where the invitation is being sent
+		validateUserPermissionsOnChannels(c, memberInvite)
```

Comment:
While it's a good first-line-of defence that we've doing this validation at invite *send* time, it's more important to do this validation on invite *receive* time, or else a user can keep a backdoor to a private channel that they know they'll be removed from.